### PR TITLE
feat(driving-license): check if 65+ renewal is possible

### PR DESCRIPTION
**Pull request by stjanilofts**
https://app.asana.com/0/1204431115744096/1208446138184156

## Summary by CodeRabbit

- **New Features**
  - Introduced a new license renewal type for individuals aged 65 and older.
  - Added specific messages for eligibility regarding extended driving licenses.
  - Enhanced eligibility checks for driving license applications, including checks for the new renewal type.
  - Added a new API method to check eligibility for the renewal process for older applicants.
  - Expanded the set of requirements with a new enum value related to driving licenses.
  - Added an optional property for issuance information in the driving license category type.

- **Bug Fixes**
  - Corrected a typographical error in an API call related to health declarations.

- **Documentation**
  - Updated messaging framework to include new eligibility criteria.

- **Chores**
  - Added new constants for extended license categories and remarks.

### CODE REVIEW

File: libs/api/domains/driving-license/src/lib/drivingLicense.service.ts [hunk 1]
```diff
@@ -355,8 +355,15 @@ export class DrivingLicenseService {
     }
   }
 
-  async canApplyFor(type: 'B-full' | 'B-temp' | 'BE', token: string) {
-    if (type === 'B-full') {
+  async canApplyFor(
+    type: 'B-full' | 'B-temp' | 'BE' | 'B-full-renewal-65',
```
Comment (kksteini) [inline libs/api/domains/driving-license/src/lib/drivingLicense.service.ts:359]: Should this be an enum/union type at this point?

File: libs/api/domains/driving-license/src/lib/drivingLicense.type.ts [hunk 2]
```diff
@@ -74,6 +74,7 @@ export enum RequirementKey {
   hasHadValidCategoryForFiveYearsOrMore = 'HasHadValidCategoryForFiveYearsOrMore',
   //TODO: Remove when RLS/SGS supports health certificate in BE license
   beRequiresHealthCertificate = 'beRequiresHealthCertificate',
+  noExtendedDrivingLicense = 'NoExtendedDrivingLicense',
```
[the remark this answers is not in the document]
Comment (kksteini) [reply libs/api/domains/driving-license/src/lib/drivingLicense.type.ts:77]: I belive line 29 in `libs/application/templates/driving-license/src/fields/EligibilitySummary/extractReasons.ts` is using this, actually.

**Review by kksteini — APPROVED** [review]
Some thoughts but otherwise LGTM :+1:

File: libs/api/domains/driving-license/src/lib/drivingLicense.service.ts (hunk 1 above)
Comment (stjanilofts) [reply libs/api/domains/driving-license/src/lib/drivingLicense.service.ts:359]: yes, might be a good idea to refactor in another PR

**Pull request merged**